share/completions/magento.fish

```diff
@@ -0,0 +1,720 @@
+end
+
+function __fish_print_magento_deploy_modes -d "Shows all available deploy modes"
+  echo developer\t"Development mode"
+  echo production\t"Production mode"
+end
+
+###
+# Retrieves all existing magento admin users from the mysql database
+###
+function __fish_print_magento_users -d "Shows all existing magento admin users"
+  set -l users (__fish_print_magento_mysql_command 'select username from admin_user;')
+
+  for i in $users
+    echo $i
+  end
+end
+
+function __fish_print_magento_cache_types -d "Shows all available cache types"
+  echo config\t"Configuration"
+  echo layout\t"Layout"
+  echo block_html\t"Block HTML output"
+  echo collections\t"Collections data"
+  echo db_ddl\t"Database schema"
+  echo eav\t"Entity attribute value (EAV)"
+  echo full_page\t"Page cache"
+  echo reflection\t"Reflection"
+  echo translate\t"Translations"
+  echo config_integration\t"Integration configuration"
+  echo config_integration_api\t"Integration API configuration"
+  echo config_webservice\t"Web services configuration"
+end
+
+function __fish_print_magento_verbosity_levels -d "Shows all available verbosity levels"
+  echo 1\t"normal output"
+  echo 2\t"more verbose output"
+  echo 3\t"debug"
+end
+
+function __fish_print_magento_list_formats -d "Shows all available output formats"
+  echo txt\t"Prints as plain text"
+  echo json\t"Prints as JSON"
+  echo xml\t"Prints as XML"
+end
+
+function __fish_print_magento_commands_list -d "Lists magento commands"
+    set -l commands help list admin:user:create admin:user:unlock app:config:dump cache:clean cache:disable cache:enable cache:flush cache:status catalog:images:resize catalog:product:attributes:cleanup cron:run customer:hash:upgrade deploy:mode:set deploy:mode:show dev:source-theme:deploy dev:tests:run dev:urn-catalog:generate dev:xml:convert i18n:collect-phrases i18n:pack i18n:uninstall indexer:info indexer:reindex indexer:reset indexer:set-mode indexer:show-mode indexer:status info:adminuri info:backups:list info:currency:list info:dependencies:show-framework info:dependencies:show-modules info:dependencies:show-modules-circular info:language:list info:timezone:list maintenance:allow-ips maintenance:disable maintenance:enable maintenance:status module:disable module:enable module:status module:uninstall sampledata:deploy sampledata:remove sampledata:reset setup:backup setup:config:set setup:cron:run setup:db-data:upgrade setup:db-schema:upgrade setup:db:status setup:di:compile setup:install setup:performance:generate-fixtures setup:rollback setup:static-content:deploy setup:store-config:set setup:uninstall setup:upgrade theme:uninstall
+    for i in $commands
+        echo $i
+    end
+end
+
+#########################################################
+
+function __fish_magento_not_in_command -d "Checks that prompt is not inside of magento command"
+    for i in (commandline -opc)
+        if contains -- $i (__fish_print_magento_commands_list)
```

Comment:
Since the command list doesn't change, you can presumably save it in a variable before the loop.